### PR TITLE
feat: support workload identity setting in static PV mount on AKS

### PR DESCRIPTION
**Pull request by cvvz**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
support workload identity setting in static PV mount on AKS
#### Which issue(s) this PR fixes:

Fixes 
https://github.com/Azure/AKS/issues/3432
https://github.com/Azure/azure-storage-fuse/issues/1049#issuecomment-1413265488

#### Special notes for your reviewer:
Documentation: https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/1566/files#diff-ba6bdf5d643d138b4a41b51808bf8041b48d4d519ebc3c511b4464d14a6ba639

#### Does this PR introduce a user-facing change?

```release-note
feat: support workload identity setting in static PV mount on AKS
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_storageaccount.go [hunk 1]
```diff
@@ -128,6 +132,65 @@ func (az *Cloud) getStorageAccounts(ctx context.Context, accountOptions *Account
...
+
+	result, err := clientFactory.NewAccountsClient().ListKeys(ctx, rgName, accountName, &armstorage.AccountsClientListKeysOptions{Expand: nil})
+	if err != nil {
+		return "", fmt.Errorf("failed to list keys, error: %w", err)
+	}
+	if result.Keys == nil {
+		return "", fmt.Errorf("failed to list keys, empty keys")
+	}
+
+	for _, k := range result.Keys {
```
Comment (andyzhangx) [inline pkg/provider/azure_storageaccount.go:180]: is it possible to reuse `GetStorageAccesskey` func?

File: pkg/provider/azure_storageaccount.go [hunk 2]
```diff
@@ -41,6 +44,7 @@ import (
 const SkipMatchingTag = "skip-matching"
 const LocationGlobal = "global"
 const privateDNSZoneNameFmt = "privatelink.%s.%s"
+const DefaultTokenAudience = "api://AzureADTokenExchange"
```
Comment (andyzhangx) [inline pkg/provider/azure_storageaccount.go:47]: this is a golint error, what's the other token Audience?

**Comment by MartinForReal** [comment]
listkeys func is added to account client https://github.com/kubernetes-sigs/cloud-provider-azure/pull/5019

File: pkg/provider/azure_storageaccount.go (hunk 2 above)
Comment (cvvz) [reply pkg/provider/azure_storageaccount.go:47]: I don't know any other Audience, for this feature, we only use `api://AzureADTokenExchange` as audience

File: pkg/provider/azure_storageaccount.go [hunk 3]
```diff
@@ -128,6 +132,65 @@ func (az *Cloud) getStorageAccounts(ctx context.Context, accountOptions *Account
 	return accounts, nil
 }
 
+// serviceAccountToken represents the service account token sent from NodePublishVolume Request.
+// ref: https://kubernetes-csi.github.io/docs/token-requests.html
+type serviceAccountToken struct {
+	APIAzureADTokenExchange struct {
+		Token               string    `json:"token"`
+		ExpirationTimestamp time.Time `json:"expirationTimestamp"`
+	} `json:"api://AzureADTokenExchange"`
```
[the remark this answers is not in the document]
Comment (cvvz) [reply pkg/provider/azure_storageaccount.go:141]: No, I've tested this feature successfully in local env. Which part do you think is a typo?

**Comment by cvvz** [comment]
> listkeys func is added to account client #5019

I've reused this func in the latest commit.

File: pkg/provider/azure_storageaccount.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (MartinForReal) [reply pkg/provider/azure_storageaccount.go:141]: `json:"api://AzureADTokenExchange"`

**Comment by MartinForReal** [comment]
@cvvz Could you please rebase the branch with master? Thanks

**Comment by MartinForReal** [comment]
related: https://github.com/kubernetes-sigs/cloud-provider-azure/pull/5126

File: pkg/provider/azure_storageaccount.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cvvz) [reply pkg/provider/azure_storageaccount.go:141]: The json structure looks like:
api://xxxxx: {
     'token': <token>,
     'expirationTimestamp': <expiration timestamp in RFC3339 format>,
},
So it's not a typo.

**Comment by cvvz** [comment]
Could you please take another look? Thanks.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/retest

**Pull request merged**